### DISCUSSION TITLE
Task 1 - 02.04  CARD TICK

### DISCUSSION THREAD

**Discussion by Andresfga** [rramosp/ai4eng.v1#79] [02 - PYTHON]
Buenas tardes profesor, tengo un problema con mi código, le creo la función para barajar y me sale Ellipsis al cargarlos y no entiendo por qué, quisiera que me ayudara haber si encuentro el error. 

def make_heaps1(c, n_heaps):
  assert n_heaps%2==1, "must have an odd number of heaps"
  assert len(c)%n_heaps==0, "the length of the deck must be a multiple of the number of heaps"
  div_c = np.array_split(c,n_heaps)
  div_per = np.random.simple(div_c)
  h=[]
  for i in div_per:
    y=list(i)
  h.append(y)
  return h
**
![image](https://user-images.githubusercontent.com/95330160/145090751-174c01fd-e23e-48d6-802e-7131ed1f6b49.png)
**

**Comment by Felipe-RA** [comment]
Saludos Andresfga

El objeto _Ellipsis_ es un objeto especial que hasta el momento sólo utiliza la librería de _numpy_.  Es utilizado como representación de las dimensiones no especificadas cuando haces un _slicing_

[Te recomiendo esta lectura para entender mejor qué función cumple el objeto Ellipsis](https://stackoverflow.com/questions/772124/what-does-the-ellipsis-object-do)


En el caso del código que acabas de pasar, no encuentro ninguna función _numpy.random.**simple**_ en la documentación de numpy, ¿a lo mejor te referías a [numpy.random.sample ] pero igualmente el código no funciona de esta ejecuta(https://numpy.org/doc/stable/reference/random/generated/numpy.random.random_sample.html#numpy.random.random_sample)?

Haz uso de la herramienta de insertar código, de esa manera no se pierde la funcionalidad del código y te podemos ayudar más fácil.

**Reply by Andresfga** [reply]
Ya logré que no me saliera el Ellipsis pero al correr el codigo lo que me returna es un h con los valores en orden de cada c en orden, encambio el ejemplo muestra que debe retornar es en la primera fila los primeros valores de c, en la segunda los segundos y asi sucesivamente, quizas el error este en la forma de bajarar las cartas con la funcion np.random.permutation pero no encuentro otra que me pueda servir, quisiera saber si existe alguna otra opcion de esciges las cartas para asignarlas con el for.

def make_heaps(c, n_heaps=3):
    assert n_heaps%2==1, "must have an odd number of heaps"
    assert len(c)%n_heaps==0, "the length of the deck must be a multiple of the number of heaps"
    div_c = np.array_split(c,n_heaps)
    div_per = np.random.permutation(div_c)
    h=[]
    for i in div_per:
      y=list(i)
      h.append(y)
    return h

![image](https://user-images.githubusercontent.com/95330160/145430528-a43ccf18-928c-4727-b3d7-c98c794ba6e0.png)

**Reply by Felipe-RA** [reply]
Ciertamente el problema está en que uses la función de random.permutations. Estas creando los heaps aleatoriamente con los elementos de la baraja cuando estos deben ser tomados avanzando según el número de cartas que tenga cada heap.

Tomando tu ejemplo anterior: 
![image](https://user-images.githubusercontent.com/49454068/145581588-8037aeed-1c1c-4187-983f-1c7b3fdde3b3.png)


Te sugiero que hagas uso del slicing para resolver este problema. 

Mirate este post para entender la notación: https://stackoverflow.com/questions/509211/understanding-slice-notation

Y este para comprenderlo en la práctica con algunos ejemplos: https://www.geeksforgeeks.org/python-list-comprehension-and-slicing/
